secure.go

```diff
@@ -437,9 +437,13 @@ func (s *Secure) isSSL(r *http.Request) bool {
 // Used by http.ReverseProxy.
 func (s *Secure) ModifyResponseHeaders(res *http.Response) error {
 	if res != nil && res.Request != nil {
-		// Fix Location response header http to https when SSL is enabled.
+		// Fix Location response header http to https when SSL is enabled
+		// And the response location header includes the SSLHost,
+		// And no port is defined in the location header.
 		location := res.Header.Get("Location")
-		if s.isSSL(res.Request) && strings.Contains(location, "http:") {
+		if s.isSSL(res.Request) &&
+			strings.Contains(location, fmt.Sprintf("http://%s", s.opt.SSLHost)) &&
```

Comment:
Theoretically, this is incorrect if the SSLHost is a prefix of a different domain, for instance SSLHost is `traefik.internal` and the Location has `https://traefik.internal.other`. 

Thus the correct way to specify this would be:

    strings.HasPrefix(location, fmt.Sprintf("http://%s/", s.opt.sslHost)) || location == fmt.Sprintf("http://%s", s.opt.sslHost)

Reply:
Also, HasPrefix is better than Contains, the sslHost might legally be present in the Location even if it is not the referred domain, for instance:

     Location: http://external.domain/path/page?referrer=http://secure.domain